libkbfs/node_cache.go

```diff
@@ -402,6 +402,50 @@ func (ncs *nodeCacheStandard) AllNodes() (nodes []Node) {
 	return nodes
 }
 
+// AllNodeChildren implements the NodeCache interface for nodeCacheStandard.
+func (ncs *nodeCacheStandard) AllNodeChildren(n Node) (nodes []Node) {
+	var rootWrappers []func(Node) Node
+	defer func() {
+		for i, n := range nodes {
+			nodes[i] = ncs.wrapNodeStandard(n, rootWrappers)
+		}
+	}()
+
+	ncs.lock.Lock()
+	defer ncs.lock.Unlock()
+	rootWrappers = ncs.rootWrappers
+	nodes = make([]Node, 0, len(ncs.nodes))
+	entryIDs := make(map[NodeID]bool)
+	for _, entry := range ncs.nodes {
```

Comment:
Is `ncs` per repo? Otherwise it seems this'd be slow for kbfs processes that have been heavily used.

Reply:
`ncs` is per-TLF, not per-repo.  But, a node's lifetime in the cache is controlled by the kernel; once the kernel forgets a node, it's removed from the cache.  So the size of each cache is limited by the number of nodes the OS is keeping about the TLF, which is presumably limited to a somewhat reasonable number.

But yes, I agree it would perform better if we keep more state connected a parent to its children.  But that's going to be a little complex, so I wanted to wait to see if performance was a problem first.  The next PR adds some debug log messages that should help figure that out.

Reply:
Sounds good. Thanks for the background!